### PR TITLE
Update CI

### PR DESCRIPTION
**Pull request by isinyaaa**
This commit updates CI hooks and separates tasks as logical steps to
facilitate overviewing failing jobs.

Signed-off-by: Isabella Basso <isabbasso@riseup.net>

### CODE REVIEW

**Comment by isinyaaa** [comment]
@rodrigosiqueira @kwy95 I'd appreciate some help on this one, as I'm not sure what's missing here.

File: .github/workflows/test_setup_and_docs.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Update the system
+        run: |
+          sudo apt-get update -y
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get install -y bash python-docutils python3-sphinx
```
Comment (rodrigosiqueira) [inline .github/workflows/test_setup_and_docs.yml:21]: I think you need `graphviz` here

File: .github/workflows/unit_tests.yml [hunk 2]
```diff
@@ -10,45 +10,25 @@ jobs:
...
 
-      - name: Unit tests
+      - name: Setup mock git user
         run: |
           git config --global user.email "kw@kworkflow.net"
           git config --global user.name "Kworkflow"
-          git init
-          git commit --allow-empty -m "Test commit 1"
-          git commit --allow-empty -m "Test commit 2"
-          ./run_tests.sh
```
Comment (rodrigosiqueira) [inline .github/workflows/unit_tests.yml:36]: I think deleting these lines caused some issues with the tests.
@kwy95 any idea here?

**Comment by rodrigosiqueira** [comment]
@isinyaaa don't forget to rebase this PR.

File: .github/workflows/unit_tests.yml (hunk 2 above)
Comment (isinyaaa) [reply .github/workflows/unit_tests.yml:36]: I managed to solve this by add more depth to the checkout action :)

**Comment by isinyaaa** [comment]
I'm still unsure about what's causing kcov to fail... There were many things installed during the setup which kcov might be using, though I'm not sure either about how it works or how to go about reading its output.

**Comment by rodrigosiqueira** [comment]
@kwy95 any idea?

File: .github/workflows/kcov.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        run: |
+          sudo apt-get update -y
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get install -y kcov shunit2 bc sqlite3 bsdmainutils
+
+      - name: Prepare for coverage test
+        run: |
+          mkdir kcov_out/
```
Comment (rodrigosiqueira) [inline .github/workflows/kcov.yml:25]: Could you try:
```
          ./run_tests.sh prepare
          mkdir kcov_out/
```

File: .github/workflows/kcov.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Update the system
+        run: |
+          sudo apt-get update -y
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get install -y kcov shunit2 bc sqlite3 bsdmainutils
```
Comment (rodrigosiqueira) [inline .github/workflows/kcov.yml:21]: Maybe you need to add `git dash`

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Hi @isinyaaa I added some ideas, could you try them?

File: .github/workflows/kcov.yml (hunk 4 above)
Comment (isinyaaa) [reply .github/workflows/kcov.yml:21]: Hm, you're right, git isn't pre-installed, but dash is actually available in the base image.

Comment (isinyaaa) [reply .github/workflows/kcov.yml]: I'm trying with dash now, for my last attempt, but I can't test this locally unfortunately as the local runner always runs as root then breaks other tests unrelated to what is happening in here.

**Comment by rodrigosiqueira** [comment]
Veryyyyy nice PR! Merged to the unstable branch :)
Thanks a lot.

**Comment by rodrigosiqueira** [comment]
Veryyyyy nice PR! Merged to the unstable branch :)
Thanks a lot.

**Pull request closed without merging**